application/classes/Logger.php

```diff
@@ -9,41 +9,46 @@ public static function instance() {
 	}
 	
 	private static function loadLogger() {
-		$prov_config = Kohana::$config->load('logger');
+		if (is_null(Kohana::$config)) {
+			$prov_config = [ 'type' => 'syslog', 'config' => 'controll' ];
+		} else {
+			$prov_config = Kohana::$config->load('logger');
+		}
 		$fullclass = "Log_" . str_replace('/', '_', ucfirst($prov_config['type']));
-		$logger = new $fullclass($prov_config);
-		Kohana::$log->attach($logger);
+		$logger = new $fullclass(array_key_exists('config', $prov_config) ? $prov_config['config'] : $prov_config);
+		if (!is_null(Kohana::$log))
+			Kohana::$log->attach($logger);
 		return $logger;
 	}
 
 	public static function debug($message, array $context = NULL, array $additional = NULL) {
 		self::instance();
-		return Kohana::$log->add(Kohana_Log::DEBUG, $message, self::parseContext($context), $additional);
+		return (Kohana::$log ?? Log::instance())->log(Kohana_Log::DEBUG, $message, self::parseContext($context));
```

Comment:
Each of these defensive checks is similar to the state issue I mentioned above and the code suffers. Lots of duplication of seemingly inane logic. Why would log not be set at this point?
To resolve these, maybe use a Null Object pattern when no logging is available or always call Log::instance (which you seem to trust implicitly as a fallback)

Note that static calls such as these are counter productive for testing purposes and do not allow you to properly mock their operation. Dependency Injection with Lazy Loading patterns can help here